cpp/src/interop/from_arrow.cu

```diff
@@ -169,7 +169,7 @@ struct dispatch_to_cudf_column {
 
 std::unique_ptr<column> get_empty_type_column(size_type size)
 {
-  return std::make_unique<column>(data_type(type_id::EMPTY), size, rmm::device_buffer{});
+  return make_empty_column(type_id::EMPTY);
```

Comment:
This change is the reason that the tests are currently failing. The Python layer relies on being able to construct an empty column with a finite size. You can see the difference with this minimal example.
Before the change:
```
(rapids) rapids@compose:~/cudf$ python -c "import cudf; print(cudf.core.column.as_column([cudf.NA]))"
<cudf.core.column.string.StringColumn object at 0x7f5da8778240>
1 nulls
dtype: object
```
And after:
```
(rapids) rapids@compose:~/cudf$ python -c "import cudf; print(cudf.core.column.as_column([cudf.NA]))"
<cudf.core.column.string.StringColumn object at 0x7f64504cc040>
0 nulls
dtype: object
```

The C++ layer has [a check preventing empty columns from having a null mask](https://github.com/rapidsai/cudf/blob/branch-22.06/cpp/src/column/column_view.cpp#L51), but it doesn't check on the null _count_. Up until now, the Python code has been relying on operating with a somewhat broken intermediate state: an empty column with nonzero size, no null mask, and an unknown null count, being constructed in a case when in fact the data is all invalid. However, you can also see that the null count is being computed correctly in the above example, so I'm not quite sure where the EMPTY column is getting converted into a reasonable string column. 

I need to do a bit more digging to figure out exactly what the best solution is here.